src/pykiso/cli.py

```diff
@@ -84,6 +89,35 @@ def check_file_extension(click_context: click.Context, param: click.Parameter, p
     return paths
 
 
+def active_threads() -> list[str]:
+    """Get the names of all active threads except the main thread."""
+    return [thread.name for thread in threading.enumerate()][1:]
+
+
+def check_and_handle_unresolved_threads(log: logging.Logger, timeout: int = 10) -> None:
+    """Check if there are unresolved threads and handle them.
+    Process for unresolved threads:
+    - If there are unresolved threads, log a warning and wait for a timeout.
+    - If the threads are still running after the timeout, log a fatal error and force exit.
+    - If the threads are properly shut down, log a warning and exit normally.
+    """
+    # Skip main thread and get running threads
+    running_threads = active_threads()
+
+    if len(running_threads) > 0:
+        for thread in running_threads:
+            log.warning(f"Unresolved thread {thread} is still running")
+        log.warning(f"Wait {timeout}s for unresolved threads to be terminated.")
+        time.sleep(timeout)
+        if threading.active_count() > 1:
+            log.fatal(
+                f"Unresolved threads {', '.join(active_threads())} are still running after {timeout} seconds. Force pykiso to Exit."
+            )
+            os._exit(test_execution.ExitCode.UNRESOLVED_THREADS)
+        else:
+            log.warning("Unresolved threads has been properly shut down. Normal exit.")
+
+
```

Comment:
This code should not be part of the cli! This should be in https://github.com/eclipse/kiso-testing/blob/master/src/pykiso/test_setup/dynamic_loader.py#L333! Please provide a fix!